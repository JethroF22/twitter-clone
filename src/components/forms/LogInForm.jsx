import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import { logIn } from '../../actions/auth';

class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  };

  render() {
    return (
      <div>
        <h3>Log In</h3>
        <form>
          {this.state.errors.authenticationError && (
            <p>{this.state.errors.authenticationError}</p>
          )}
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <button type="submit">Log In</button>
        </form>
      </div>
    );
  }
}

export default LogInForm;
