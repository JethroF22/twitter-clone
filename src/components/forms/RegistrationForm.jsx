import React, { Component } from 'react';

class RegistrationForm extends Component {
  state = {
    username: '',
    password: '',
    confirmPassword: '',
    handle: '',
    email: ''
  };

  render() {
    return (
      <div>
        <h3>Create an account</h3>
        <form onSubmit={() => console.log('submitted')}>
          <label htmlFor="username">Username: </label>
          <input
            type="text"
            name="username"
            value={this.state.username}
            onChange={() => console.log('Username changed')}
          />
          <br />
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="handle">Handle: </label>
          <input
            type="text"
            name="handle"
            value={this.state.handle}
            onChange={() => console.log('Handle changed')}
            placeholder="Example: @i_am_jerry"
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <label htmlFor="confirmPassword">Confirm Password: </label>
          <input
            type="password"
            name="confirmPassword"
            value={this.state.confirmPassword}
            onChange={() => console.log('Confirm Password changed')}
          />
          <br />
        </form>
      </div>
    );
  }
}

export default RegistrationForm;
