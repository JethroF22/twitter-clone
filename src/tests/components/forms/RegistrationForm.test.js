import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';

import { RegistrationForm } from '../../../components/forms/RegistrationForm';

import { users } from '../../seed/seed';

let wrapper, name, registerUserSpy, user, state;

describe('Registration Form', () => {
  beforeEach(() => {
    registerUserSpy = jest.fn().mockResolvedValueOnce(null);
    user = users[0];
    state = { ...user, confirmPassword: user.password, errors: {} };
    wrapper = shallow(<RegistrationForm registerUser={registerUserSpy} />);
    wrapper.setState(state);
    name = 'Jethro';
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should handle onChange', () => {
    wrapper.find('input[name="name"]').prop('onChange')({
      target: {
        name: 'name',
        value: name
      }
    });
    expect(wrapper.state('name')).toBe(name);
  });

  test('should handle onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(wrapper.state('errors')).toEqual({});
    expect(registerUserSpy).toHaveBeenLastCalledWith({
      name: user.name,
      handle: user.handle,
      email: user.email,
      password: user.password
    });
  });

  test('should display validation errors', () => {
    state = {
      ...state,
      name: 'Jeff'
    };
    const nameError = 'Name must be at least 6 characters';
    wrapper.setState(state);
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(wrapper.state('errors')).toEqual({
      name: nameError
    });
    expect(
      wrapper
        .find('p')
        .at(0)
        .text()
    ).toBe(nameError);
  });
});
