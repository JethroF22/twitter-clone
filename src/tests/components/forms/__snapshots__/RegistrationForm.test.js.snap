// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form should render correctly 1`] = `
<div>
  <h3>
    Create an account
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value="Jethro"
    />
    <br />
    <label
      htmlFor="email"
    >
      Email: 
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="email"
      value="jethro@gmail.com"
    />
    <br />
    <label
      htmlFor="handle"
    >
      Handle: 
    </label>
    <input
      name="handle"
      onChange={[Function]}
      placeholder="Example: @i_am_jerry"
      type="text"
      value="iamjethro"
    />
    <br />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="password1"
    />
    <br />
    <label
      htmlFor="confirmPassword"
    >
      Confirm Password: 
    </label>
    <input
      name="confirmPassword"
      onChange={[Function]}
      type="password"
      value="password1"
    />
    <br />
    <button
      disabled={false}
      type="submit"
    >
      Create Account
    </button>
  </form>
</div>
`;
